### ISSUE TITLE
The replacement options for 'Does not move' in the ship editor on newer builds cannot be disabled via the 'immoble' ship flag.

### ISSUE THREAD

**Issue by DefCynodont** [scp-fs2open/fs2open.github.com#6489]
Newer build replace the 'Does not move' option in the ship editor with two separate `Does not change position` and `Does not change orientation` settings. as far as I know, these settings are set to true automatically if a mission loads in FRED with the `Does not move` ('immoble' ship flag) option checked on a ship. 

However, it is not 1 to 1 with the 'immoble' ship flag, as using SEXPs to change the 'immoble' ship flag for ships that start with the replacement settings are not effected by it.

That can and has caused issues with missions that start with a ship immobilized and free them later. as these ships will no longer move when the mission uses `set-ship-flag` to enable their movement again. (and there are a few examples I can think of where a ship may by set as immobilized at mission start)

Attached is a test mission that demonstrates this using Morning Stars and unkillable containers. 
One container (the left one) has these settings set as true, and the right one is normal, pressing 1 will set the `immoble` flag as false for the left container, and true for the right container, pressing 2 sets the flag as false again for the right container. not that while the right container responds to the `immoble` flag as expected, the left one stays still regardless of the flag's setting.
[moving things test mission.zip](https://github.com/user-attachments/files/18256196/moving.things.test.mission.zip)

**Comment by Goober5000** [comment]
In short, this is expected and unavoidable behavior.

The "immobile", "don't-change-position", and "don't-change-orientation" flags are all independent ship flags.  Since using `alter-ship-flag` operates on the flags themselves, setting or clearing one doesn't set or clear the others.  So, using overlapping flags will cause overlapping behaviors like this.  What you can do, instead, is use the `set-mobile` and `set-immobile` SEXPs.  These will clear the "don't-change" flags to prevent conflicts.

I'll quote from PR #6304:

> In the FRED and qtFRED ship flags dialogs, the "immobile" flag (aka "Does Not Move") is replaced by the two new flags, but in FSO the situation is more complicated. The old and new flags coexist and have overlapping functionality, so the mission designer must be careful to set and unset the appropriate flag (that is, it is possible for both "immobile" and "don't-change-position" to be set at the same time, in which case the ship cannot change position without both flags being unset). The `set-immobile` and `set-mobile` legacy SEXPs ensure that the new flags are removed so as not to conflict with the legacy flag, but the `alter-ship-flag` and `are-ship-flags-set` SEXPs and the `getFlag` and `setFlag` script functions use exactly what the FREDder/scripter specifies and nothing else.
> 
> (I attempted to completely replace "immobile" with the two newer flags, but it proved intractable to do this while maintaining reverse compatibility: retrofitting single-flag functions to apply dual flags for one particular situation made the functions harder to understand as well as introducing many edge cases and opportunities for bugs or "gotchas".)

Basically, the old flag can't be deprecated and the new flags can't replace the old flag due to numerous problems that this causes.  So, the only solution is to retrofit old operators where possible (such as `set-mobile`/`set-immobile`) while documenting the potential gotchas.

**Comment by naomimyselfandi** [comment]
I understand that perspective, @Goober5000, but I also understand the perspective of someone who just spent hours tracking down a mission bug that came up out of nowhere after they fixed a bug that had nothing to do with the `immobile` flag. Documenting this stuff in the operators is a good idea, but it can only get you so far. Someone who's maintaining a mission won't be thinking about this change when they open a mission; they may not even remember that the mission has `immobile` objects. For that reason, I feel that updating these flags automatically (or more generally, any sort of automatic rewriting without identical semantics), without warning, is a pretty bad footgun.

In my mind, this is a human problem, not a technical one; the solution isn't to hack a special case into `alter-ship-flag` or what have you, but to provide a warning before it burns someone. When the `immobile` flag is replaced, check if the mission mentions the `immobile` flag in any of its events, and if so, show a warning (or unconditionally show the warning when the flag is replaced, if that logic is too thorny).

**Comment by Goober5000** [comment]
I mean, I'm certainly sympathetic to the FREDer perspective.  That's why I added extra documentation to the `set-mobile`/`set-immobile` SEXP operators.  Keep in mind that splitting the "immobile" flag into two flags was itself a FREDer request.

Yeah, the case where an old mission is opened, silently migrated without the FREDer's knowledge, and then re-saved would definitely be incredibly frustrating.  I can certainly add a check for that.  It might also be possible to add a more general check that would look for uses of `alter-ship-flag` that conflict with flags set in the ship editor.
